sky/resources.py

```diff
@@ -267,6 +280,34 @@ def disk_size(self) -> int:
     def image_id(self) -> Optional[Dict[str, str]]:
         return self._image_id
 
+    @property
+    def disk_type(self) -> str:
+        return self._disk_type
+
+    def get_cloud_disk_type_str(self) -> str:
+        if isinstance(self.cloud, clouds.GCP):
+            return ('pd-ssd' if self.disk_type == 'high' else 'pd-balanced'
+                    if self.disk_type == 'medium' else 'pd-standard')
+        if isinstance(self.cloud, clouds.AWS):
+            return ('io2' if self.disk_type == 'high' else
+                    'gp3' if self.disk_type == 'medium' else 'gp3')
+        if isinstance(self.cloud, clouds.Azure):
+            return (
+                # TODO(tian): maybe use UltraSSD_LRS?
+                'Premium_LRS' if self.disk_type == 'high' else 'StandardSSD_LRS'
+                if self.disk_type == 'medium' else 'Standard_LRS')
+        raise NotImplementedError(
+            f'Disk type of cloud {self.cloud} is not implemented yet.')
```

Comment:
I'm wondering whether it is feasible to add `specify_disk_type` in `requested_features` since other features are all task-level features and this is a resource-level feature. That is to say, multi resources in the same task may not require this feature simultaneously. 🤔
Currently, I move this check into `_try_validate_disk_type`.